### PR TITLE
Formula- and term-based Wald-tests

### PR DESCRIPTION
**Pull request by MartinStancsicsQC**
Checklist
* [x] Added a `CHANGELOG.rst` entry

This PR uses the new functionality in glum v3 (namely the formula interface and the existence of term names) to extend the interface for the `wald_test` method. The latter now also accepts a `term` or a `formula` parameter.

### CODE REVIEW

File: src/glum/_glm.py
Comment (MatthiasSchmidtblaicherQC) [inline src/glum/_glm.py:1945]: See above.

Comment (MatthiasSchmidtblaicherQC) [reply src/glum/_glm.py:1945]: It is probably worth expanding in docstring of `wald_test` how categoricals that are never encoded as dummies are handled. This seems like a distinguishing feature of glum's wald tests.

**Review by MatthiasSchmidtblaicherQC — CHANGES_REQUESTED** [review]

File: src/glum/_glm.py
Comment (MartinStancsicsQC) [reply src/glum/_glm.py:1532]: Yes, exactly. I'll try to make it clearer.

Comment (MartinStancsicsQC) [reply src/glum/_glm.py]: How about [this](https://github.com/Quantco/glum/blob/89a5ba4ae7f9b7861288f59670b41c551c3f9cb5/src/glum/_glm.py#L1532-L1538)?

**Review by MarcAntoineSchmidtQC — APPROVED** [review]
Looks good and well tested. Happy to merge this when you are ready.

File: src/glum/_glm.py
Comment (MatthiasSchmidtblaicherQC) [reply src/glum/_glm.py]: Looks good!

**Review by MatthiasSchmidtblaicherQC — APPROVED** [review]

**Comment by MartinStancsicsQC** [comment]
Great, I'll merge it right after the formula PR is merged.

**Pull request merged**
